packages/integration-sdk-runtime/src/synchronization/__tests__/index.test.ts

```diff
@@ -59,6 +59,30 @@ describe('initiateSynchronization', () => {
     });
   });
 
+  test('starts a synchronization job with integration job id if provided', async () => {
+    const mockIntegrationJobId = 'test-integration-job-id';
+    const mockSyncJob = generateSynchronizationJob();
+    mockSyncJob.integrationJobId = mockIntegrationJobId;
+
+    const context = createTestContext();
+    context['integrationJobId'] = mockIntegrationJobId;
+    const { apiClient } = context;
+
+    const postSpy = jest.spyOn(apiClient, 'post').mockResolvedValue({
+      data: { job: mockSyncJob },
+    });
+
+    const synchronizationContext = await initiateSynchronization(context);
+
+    expect(synchronizationContext.job).toEqual(mockSyncJob);
+    expect(postSpy).toHaveBeenCalledTimes(1);
+    expect(postSpy).toHaveBeenCalledWith('/persister/synchronization/jobs', {
```

Comment:
nit: might be interesting to also add an assertion on the parameters on logger.child()
We want the integrationJobId passed into the logger to be the same as the one passed into the post method, right?